app/tracing.ts

```diff
@@ -0,0 +1,227 @@
+import { BatchSpanProcessor, SamplingDecision, SamplingResult } from '@opentelemetry/sdk-trace-base';
+import { Resource } from '@opentelemetry/resources';
+import { SemanticResourceAttributes } from '@opentelemetry/semantic-conventions';
+import { NodeTracerProvider } from '@opentelemetry/sdk-trace-node';
+import { registerInstrumentations } from '@opentelemetry/instrumentation';
+import { getNodeAutoInstrumentations } from '@opentelemetry/auto-instrumentations-node';
+import { OTLPTraceExporter } from '@opentelemetry/exporter-trace-otlp-http';
+import { HttpInstrumentation } from '@opentelemetry/instrumentation-http';
+import { AWSXRayIdGenerator } from '@opentelemetry/id-generator-aws-xray';
+import { Attributes, Context, Link, SpanKind, trace } from '@opentelemetry/api';
+import { AWSXRayPropagator } from '@opentelemetry/propagator-aws-xray';
+import env from './util/env';
+import { random } from 'lodash';
+import { AwsInstrumentation } from '@opentelemetry/instrumentation-aws-sdk';
+
+
+
+// Register instrumentation libraries for Node.js auto-instrumentation and capturing of
+// outgoing HTTP calls
+registerInstrumentations({
+  instrumentations: [
+    getNodeAutoInstrumentations(),
+    new HttpInstrumentation(),
+    new AwsInstrumentation({}),
+  ],
+});
+
+const resource =
+  Resource.default().merge(
+    new Resource({
+      [SemanticResourceAttributes.SERVICE_NAME]: env.harmonyClientId,
+      [SemanticResourceAttributes.SERVICE_VERSION]: '0.1.0',
+    }),
+  );
+interface SampleCheckContext {
+  context?: Context,
+  traceId?: string,
+  spanName?: string,
+  spanKind?: SpanKind,
+  attributes?: Attributes,
+  links?: Link[],
+}
+
+// maximum number of rejected traces we will track to avoid sending partial traces
+const REJECTED_TRACE_CACHE_LIMIT = 10000;
+
+// class used to track to the traces we have rejected so we don't send subsegments that have no
+// parent (due to the parent being rejected).
+// TODO This should probably be tracked externally in
+// Redis or some other cache so that it works across Harmony instances, but this is better than
+// nothing for now.
+class RejectedTraceMonitor {
+  rejectedTraces: string[];
+
+  constructor() {
+    this.rejectedTraces = [];
+  }
+
+  reject(traceId: string): void {
+    this.rejectedTraces.unshift(traceId);
+    if (this.rejectedTraces.length > REJECTED_TRACE_CACHE_LIMIT) {
+      this.rejectedTraces.pop();
+    }
+  }
+
+  isRejected(traceId: string): boolean {
+    return this.rejectedTraces.includes(traceId);
+  }
+
+}
+
+const rejectedTraces = new RejectedTraceMonitor();
+
+/**
+ * Test function that checks a single criteria to see if we want to sample a given trace/span
+ * @param context - the context of the trace to be (or not to be) sampled
+ */
+type _SampleCheckFn = (context: SampleCheckContext) => SamplingDecision;
+
+const ELB_HEALTH_CHECK_AGENT = 'ELB-HealthChecker/2.0';
+
+/**
+ * {@inheritDoc SampleCheckFn}
```

Comment:
This is new but I think I get the intention. Is it an issue with any tooling that we define `_SampleCheckFn` (with an underscore) not `SampleCheckFn`?

Reply:
Yeah - this if wrong. I should have updated these to `_SampleCheckFn` when I changed the actual function to avoid linter complaints about unused functions. Fixed.